File: conformance/tests/bounded-loop/bounded-loop.tsop.go
package contract

import "tsop"

type BoundedLoop struct {
	tsop.SmartContract
	ExpectedSum tsop.Int `tsop:"readonly"`
}

func (c *BoundedLoop) Verify(start tsop.Int) {
	sum := tsop.Int(0)
	for i := tsop.Int(0); i < 5; i++ {
		sum = sum + start + i
	}
	tsop.Assert(sum == c.ExpectedSum)
}

File: conformance/tests/bounded-loop/bounded-loop.tsop.rs
use tsop::prelude::*;

#[tsop::contract]
struct BoundedLoop {
    #[readonly]
    expected_sum: Int,
}

#[tsop::methods(BoundedLoop)]
impl BoundedLoop {
    #[public]
    fn verify(&self, start: Int) {
        let mut sum: Int = 0;
        for i in 0..5 {
            sum = sum + start + i;
        }
        assert!(sum == self.expected_sum);
    }
}
